### PR TITLE
Add grunt

### PR DESCRIPTION
**Pull request by swansong**
Added Grunt with some favorite modules and changed the structure of the project slightly. Some JSHint errors are there, but I figured I should just force the concat and minify in the meantime.

### CODE REVIEW

**Comment by toastynerd** [comment]
Yeah, I used a similar structure in my pr https://github.com/austinhinderer/Player/pull/5 I also ditched the individual file closures and added a footer and header that essentially wrapped all of it in one closure (Angular uses this strat). Although for some reason mine isn't merging right now. I like the addition of uglify but I think we should have a separate "production" build task, rather than generating it with the dev build. That's my 2 cents.

**Comment by austinhinderer** [comment]
I'm good with shipping a uglified file for development as long as we ship sourcemaps. Thoughts?

**Comment by austinhinderer** [comment]
Also, I'm a big fan of SCSS. I'd prefer to use that for CSS management.

File: gruntfile.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+     // },
+      js: {
+        files: ['<%= concat.dist.src %>'],
+        tasks: ['js']
+      }
+    }
+  });
+
+  grunt.loadNpmTasks('grunt-notify');
+  //grunt.loadNpmTasks('grunt-contrib-less');
```
Comment (austinhinderer) [inline gruntfile.js]: Would prefer compass + SCSS for this.

**Comment by austinhinderer** [comment]
Also, as a general rule, let's use an interactive rebase to squash down to single commits. Helps to maintain a cleaner DAG IMO OMGLOLBBQ.

**Comment by swansong** [comment]
I opened the pull request before I saw PR5 which does much of the same stuff, so let's just go with PR5 as a starting point. I'll toy with the grunt config for compass and add it to that gruntfile (along with uglify). Not really used source maps yet but it looks great, so yes to that.

What are you suggesting with the interactive rebasing? One commit per feature?

**Comment by austinhinderer** [comment]
"What are you suggesting with the interactive rebasing? One commit per feature?"

Yes, exactly.

**Pull request closed without merging**